lib/src/client.dart

```diff
@@ -3958,3 +3984,30 @@ class _EventPendingDecryption {
 
   _EventPendingDecryption(this.event);
 }
+
+enum InitState {
+  /// Initialization has been started. Client fetches information from the database.
+  initializing,
+
+  /// The database has been updated. A migration is in progress.
+  migratingDatabase,
+
+  /// The encryption module will be set up now. For the first login this also
+  /// includes uploading keys to the server.
+  settingUpEncryption,
+
+  /// The client is loading rooms, device keys and account data from the
+  /// database.
+  loadingData,
```

Comment:
we could make it more fancy and separate this out into loading rooms, device keys, etc

Reply:
For this we already have the futures we can await. Also those futures are actually done in parallel and not one after another

Reply:
Yes, but I would like to listen to one state and know what is going on instead of having to worry about the futures. Do you think this makes sense? 🤔

Reply:
It does not work with the parallel stuff. We are awaiting the futures here but they are actually parallel

Reply:
figured, thanks for confirming!